### PR TITLE
Fix parsing of record & list bodies.

### PR DESCRIPTION
**Pull request by BigCrow**
### Brief Description
Change the node structure of the record & list bodies to better represent the actual syntax of the language. This fixes some bugs as well as make the syntax more consistent & easier to read.

### Fixes
- TS can't properly detect durantion values in records. It is instead deteced as integer & a command node that create an error:
```nu
{
    key: 5sec
}
```
- TS can't properly detect when value & key begins & allow invald syntax with value & key having no space or any separator in between. The following example is considered valid by TS:
```nu
{
    key: 1536another_key: 8549
}
```

### What is changed
- Records use a new body structure which manages entries & separators in between. This fixes the separator bugs described above & creates new nodes that can be used for highlighting & indenting.
- Lists also have their structure changed in the same manner (both records & list body acts the same in nushell syntax, with exception to what is actually a record entry ``key: value`` vs a list entry ``value``.
- Record tests are added for the bugs described above. All tests have been updated with the new node syntax.

### Details
- New function is added to grammar js for general body structure. It consist of repeated statement all seperated by a mandotory separator, followed by a final statement with an optional separator.
- This function is applied to the bodies of lists & records with the corresponding entries.
- List has an added node for the list_entry itself (before it was just hardcoded into the list rule)
- Separator is added for entries ``[,\s\n\r]``
- Comment is added on top of the code to improve LSP parsing of code with tsserver (commonly used by other TS projects for grammar.sj)
- Tests are updated with new new node syntax - I've individually checked the diffs of all tests to make sure they are working as indented
- New test has been added to check if records can accept duration input.
- New test has been added to test if records accept invaild input with no separator between value & key.

### CODE REVIEW

**Review by fdncred — COMMENTED** [review]
I think this mostly looks good but what I don't understand is why the closure-xxx-yyyy tests were removed, except for 007. I expected them to be changed but not removed. Same for export-env-001-smoke-test.

**Comment by BigCrow** [comment]
It must have been a mistake on my part. Thanks for catching that - I re added the lost tests.

File: test/corpus/expr/closure.nu [hunk 1]
```diff
@@ -163,49 +172,22 @@ closure-007-record-value
...
-                  (val_number))))))
-        (record_entry
-          (identifier)
-          (val_closure
-            (pipeline
-              (pipe_element
-                (val_string)))))))))
-
-=====
-closure-008-while
```
Comment (fdncred) [inline test/corpus/expr/closure.nu:186]: was closure-008-while just left out because it's a dupe or was this accidental?

Comment (BigCrow) [reply test/corpus/expr/closure.nu:186]: Accidental thanks - I will go through the diff once again to make sure I haven't forgotten anything. But none of the test should be removed, they should only get changes.

Comment (BigCrow) [reply test/corpus/expr/closure.nu:186]: I think everything should be fine now - I went through the test diffs & I don't see anymore lost tests.

**Comment by fdncred** [comment]
Thanks

**Pull request merged**